src/redux/books/booksSlice.js

```diff
@@ -1,32 +1,60 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const KEY_API = 'GCBOOjVHPY5Sbf74W64o';
+const API_URL = `https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${KEY_API}/books`;
 
 const initialState = {
-  books: // Initial state:
-    [
-      {
-        id: 'item1',
-        title: 'The Great Gatsby',
-        description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
-        author: 'John Smith',
-        category: 'Fiction',
-      },
-      {
-        id: 'item2',
-        title: 'Anna Karenina',
-        description: 'At est officia autem culpa amet quos deserunt dolores debitis!',
-        author: 'Leo Tolstoy',
-        category: 'Fiction',
-      },
-      {
-        id: 'item3',
-        title: 'The Selfish Gene',
-        description: 'Commodi porro enim dignissimos velit, explicabo quisquam.',
-        author: 'Richard Dawkins',
-        category: 'Nonfiction',
-      },
-    ],
+  books: [],
+  isLoading: true,
+  error: '',
 };
 
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async (thunkAPI) => {
+  try {
+    const resp = await axios.get(API_URL);
+    return resp.data;
+  } catch (err) {
+    return thunkAPI.rejectWithValue('Oooops, something went wrong');
+  }
+});
+
+export const addNewBook = createAsyncThunk('book/addNewBook',
+  async (newBook, thunkAPI) => {
+    try {
+      const resp = await axios.post(API_URL, {
+        item_id: newBook.item_id,
+        title: newBook.title,
+        author: newBook.author,
+        category: newBook.category,
+      }, {
+        headers: {
+          'Content-type': 'application/json; charset=UTF-8',
+        },
+      });
+      return { message: resp, book: newBook };
+    } catch (error) {
+      return thunkAPI.rejectWithValue('Oooops, something went wrong');
+    }
+  });
+
+export const deleteBook = createAsyncThunk('book/deleteBook',
+  async (id, thunkAPI) => {
+    const url = `${API_URL}/${id}`;
+    try {
+      const resp = await axios.delete(url, {
+        item_id: id,
+      }, {
+        headers: {
+          'Content-type': 'application/json; charset=UTF-8',
+        },
+      });
+      return { message: resp, id };
+    } catch (error) {
+      return thunkAPI.rejectWithValue('Oooops, something went wrong');
+    }
+  });
```

Comment:
- [ ] I noticed a little issue with your deletebook function in the video I've shared below. The deletion process works fine, but it seems that the UI doesn't update accordingly. Additionally, if I click the delete button again, the book disappears visually, even though in the network tab it's evident that the book was already deleted. Could you please take a look and ensure that your UI updates as expected?

- [ ] I'd also like to mention a quirky error message I spotted in the console while trying to add and remove books. It's possible that this error might be related to the deletion problem mentioned earlier. Would you mind checking into this? 


   Here's the video for a better understanding.
   
   [scrnli_24_8_2023_18-45-27.webm](https://github.com/kambereBr/bookstore/assets/107952724/fb4193b0-63d4-4810-a5b2-fb9532651558)

Reply:
Thanks dear @taldr27. Now, All issues fixed 👏